pkg/plugin/plugin.go

```diff
@@ -105,18 +106,18 @@ func (p *plugin) GenerateARM(ctx context.Context, cs *api.OpenShiftManagedCluste
 
 func (p *plugin) InitializeCluster(ctx context.Context, cs *api.OpenShiftManagedCluster) error {
 	log.Info("initializing cluster")
-	initializer := initialize.NewSimpleInitializer(p.entry)
+	initializer := initialize.NewSimpleInitializer(p.entry, p.config)
 	return initializer.InitializeCluster(ctx, cs)
 }
 
 func (p *plugin) HealthCheck(ctx context.Context, cs *api.OpenShiftManagedCluster) error {
 	log.Info("starting health check")
-	healthChecker := healthcheck.NewSimpleHealthChecker(p.entry)
+	healthChecker := healthcheck.NewSimpleHealthChecker(p.entry, p.config)
 	return healthChecker.HealthCheck(ctx, cs)
 }
 
-func (p *plugin) Update(ctx context.Context, cs *api.OpenShiftManagedCluster, azuredeploy []byte) error {
+func (p *plugin) Update(ctx context.Context, cs *api.OpenShiftManagedCluster, azuredeploy []byte, config api.PluginConfig) error {
```

Comment:
It gets messy because of the initializer. My initial plan was to add a Plugin api GetPluginConfig() to the Plugin and the initilizer, but @kargakis asked me to just add an arguement to this function.

Reply:
@asalkeld why can't you do https://gist.github.com/jim-minter/d1a266c870797a2b94a826b3218fdd7a ?

Reply:
Have a look at this commit https://github.com/openshift/openshift-azure/pull/492/commits/514e71bf2a386538929d64f325273c2315db1ffc
Imagine the reverse of that.
The real issue is the Initializer. We call Deploy and Update with either a plugin or an initilizer instance. So your example is one of the easiest cases, when we just have the api instance then we don't have access to the private "config" so I initially wrote an accessor GetPluginConfig() that I placed on the Initializer, Updater and Plugin so you could safely call it anywhere.
see: https://github.com/openshift/openshift-azure/pull/492/commits/514e71bf2a386538929d64f325273c2315db1ffc#diff-c5388788ebacab203d51d71bbf0abf3bL22

Reply:
I'm still not getting it :(  Got a sec to catch up on bluejeans in the next few hours?